src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -3655,5 +3658,496 @@ private static ImmutableArray<Symbol> GetProperties(CSharpCompilation comp, stri
+            Assert.Equal(2, yRef.Length);
+            OutVarTests.VerifyModelForOutVar(model, yDecl, yRef[0]);
+            OutVarTests.VerifyNotAnOutLocal(model, yRef[1]);
+
+            var x = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "X").ElementAt(1);
+            Assert.Equal("Test(X, out var y)", x.Parent!.Parent!.Parent!.ToString());
+
+            var symbol = model.GetSymbolInfo(x).Symbol;
+            Assert.Equal(SymbolKind.Parameter, symbol!.Kind);
+            Assert.Equal("System.Int32 X", symbol.ToTestDisplayString());
+            Assert.Equal("C..ctor(System.Int32 X)", symbol.ContainingSymbol.ToTestDisplayString());
+            Assert.Same(symbol.ContainingSymbol, model.GetEnclosingSymbol(x.SpanStart));
+            Assert.Contains(symbol, model.LookupSymbols(x.SpanStart, name: "X"));
+            Assert.Contains("X", model.LookupNames(x.SpanStart));
+
+            var y = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "y").First();
+            Assert.Equal("y", y.Parent!.ToString());
+            Assert.Equal("(Test(X, out var y), y)", y.Parent!.Parent!.ToString());
+            Assert.Equal("Base(Test(X, out var y), y)", y.Parent!.Parent!.Parent!.ToString());
+
+            symbol = model.GetSymbolInfo(y).Symbol;
+            Assert.Equal(SymbolKind.Local, symbol!.Kind);
+            Assert.Equal("System.Int32 y", symbol.ToTestDisplayString());
+            Assert.Equal("C..ctor(System.Int32 X)", symbol.ContainingSymbol.ToTestDisplayString());
+            Assert.Same(symbol.ContainingSymbol, model.GetEnclosingSymbol(x.SpanStart));
+            Assert.Contains(symbol, model.LookupSymbols(x.SpanStart, name: "y"));
+            Assert.Contains("y", model.LookupNames(x.SpanStart));
+
+            var test = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "Test").First();
+            Assert.Equal("(Test(X, out var y), y)", test.Parent!.Parent!.Parent!.ToString());
+
+            symbol = model.GetSymbolInfo(test).Symbol;
+            Assert.Equal(SymbolKind.Method, symbol!.Kind);
+            Assert.Equal("System.Int32 C.Test(System.Int32 x, out System.Int32 y)", symbol.ToTestDisplayString());
+            Assert.Equal("C", symbol.ContainingSymbol.ToTestDisplayString());
+            Assert.Contains(symbol, model.LookupSymbols(x.SpanStart, name: "Test"));
+            Assert.Contains("Test", model.LookupNames(x.SpanStart));
+        }
+
+        [Fact]
+        public void BaseArguments_03()
+        {
+            var src = @"
+using System;
+
+class Base
+{
+    public Base(int X, int Y)
+    {
+    }
+
+    public Base() {}
+}
+
+record C : Base(X, Y)
+{
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (13,16): error CS8861: Unexpected argument list.
```

Comment:
Ideally this diagnostic would say something like "only a record with positional parameters may have a constructor initializer." We can follow up in #44826

Reply:
> Ideally this diagnostic would say something like "only a record with positional parameters may have a constructor initializer." We can follow up in #44826

This is preexisting condition.

---
In reply to: [436066301](https://github.com/dotnet/roslyn/pull/44876#discussion_r436066301) [](ancestors = 436066301)